entity-framework/core/querying/related-data.md

```diff
@@ -97,7 +97,9 @@ You can also configure split queries as the default for your application's conte
 
 When split queries are configured as the default, it is still possible to configure specific queries to execute as single queries:
 
-[!code-csharp[Main](../../../samples/core/Querying/RelatedData/Sample.cs?name=AsSplitQuery&highlight=5)] 
+[!code-csharp[Main](../../../samples/core/Querying/RelatedData/Sample.cs?name=AsSplitQuery&highlight=5)]
+
+By default, EF Core emits a warning if it detects single queries loading collection includes that may cause a performance issue. If you're confident that there are no related performance issue, you can explicitly configure the splitting behavior to SplitQuery to prevent these warnings from being generated.
```

Comment:
Warning is not working that way.
We issue a warning if behavior is not specified globally or in the query using operator.
No warning if user explicitly asks for AsSingleQuery.
And to suppress a warning, we need to show the warning configuration in OnConfiguring.

Reply:
Oops, wrote SplitQuery instead of SingleQuery. Is it OK after this fix? If not do you want to make a suggestion?

Reply:
This still does not clarify what is the warning, why it is issued in first place and how to suppress the warning.

Reply:
I've reworded the note, hopefully you'll find it better. If not, please make another suggestion.

> And to suppress a warning, we need to show the warning configuration in OnConfiguring.

I don't think it's necessary to show multiple ways to make the warning go away (set SingleQuery globally/on the query, warning config in OnConfiguring), from the user's perspective it's the same. In addition, configuring warnings in OnConfiguring is a global matter that should be documented elsewhere, rather than repeating this information everywhere where a warning is discussed.

Reply:
You can add a global docs somewhere about how to suppress warnings. The unrepeated part which is most crucial for user is what EventId to use to suppress it and that is going to be varying between different scenarios.
Also setting SingleQuery mode is not the (right or only) way to suppress the warning.

I will update this page myself later. You can merge this.

Reply:
> You can add a global docs somewhere about how to suppress warnings

Opened #2615 to track.

> The unrepeated part which is most crucial for user is what EventId to use to suppress it and that is going to be varying between different scenarios

Shouldn't the EventIds be discoverable from the warning itself, rather than having to make the docs more verbose/heavy with them? Typically a user will want to suppress a warning after they see it in the log, that should contain the EventId etc.

> Also setting SingleQuery mode is not the (right or only) way to suppress the warning.

Looking forward to seeing what you propose.